core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * 
+ */
+package com.netflix.conductor.core.execution.tasks;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.common.metadata.tasks.Task.Status;
+import com.netflix.conductor.common.run.Workflow;
+import com.netflix.conductor.core.events.ScriptEvaluator;
+import com.netflix.conductor.core.execution.WorkflowExecutor;
+import com.netflix.conductor.core.execution.mapper.DecisionTaskMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.script.ScriptException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Manan
+ *
+ */
+public class DoWhile extends WorkflowSystemTask {
+
+	public DoWhile() {
+		super("DO_WHILE");
+	}
+
+	Logger logger = LoggerFactory.getLogger(DecisionTaskMapper.class);
+	Map<String, Integer> iterations = new HashMap<>();
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean execute(Workflow workflow, Task task, WorkflowExecutor provider) {
+		
+		boolean allDone = true;
+		boolean hasFailures = false;
+		StringBuilder failureReason = new StringBuilder();
+		List<String> loopOver = (List<String>) task.getInputData().get("loopOver");
+
+		int iteration = iterations.get(task.getTaskId()) == null ? 0 : iterations.get(task.getTaskId());
```

Comment:
We are maintaining "iterations" map in-memory and updating the iteration number of each task. 
But since the map is in-memory, when the task is scheduled and dequeued in other JVMs, the execution history will be overwritten. Also, consider the case when a JVM goes down, we will lose execution history. 

IMO. The iteration number of each task should be a member of task instance, not DoWhile Task.